File: src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/DiskState.java
package com.raoulvdberge.refinedstorage.apiimpl.network.node;

public enum DiskState {
    NONE,
    NORMAL,
    DISCONNECTED,
    NEAR_CAPACITY,
    FULL;

    public static final int DISK_NEAR_CAPACITY_THRESHOLD = 75;

    public static DiskState get(int stored, int capacity) {
        if (stored == capacity) {
            return FULL;
        } else if ((int) ((float) stored / (float) capacity * 100F) >= DISK_NEAR_CAPACITY_THRESHOLD) {
            return NEAR_CAPACITY;
        } else {
            return NORMAL;
        }
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/apiimpl/network/node/DiskState.kt
package com.refinedmods.refinedstorage.apiimpl.network.node



enum class DiskState {
    NONE, NORMAL, DISCONNECTED, NEAR_CAPACITY, FULL;

    companion object {
        const val DISK_NEAR_CAPACITY_THRESHOLD = 75
        operator fun get(stored: Int, capacity: Int): DiskState {
            return if (stored == capacity) {
                FULL
            } else if ((stored.toFloat() / capacity.toFloat() * 100f).toInt() >= DISK_NEAR_CAPACITY_THRESHOLD) {
                NEAR_CAPACITY
            } else {
                NORMAL
            }
        }
    }
}